vault/identity_store_oidc.go

```diff
@@ -607,13 +597,13 @@ func (i *IdentityStore) pathOIDCGenerateToken(ctx context.Context, req *logical.
 	now := time.Now()
 	idToken := idToken{
 		Issuer:   config.effectiveIssuer,
-		Subject:  te.EntityID,
+		Subject:  req.EntityID,
 		Audience: role.RoleID,
 		Expiry:   now.Add(role.TokenTTL).Unix(),
 		IssuedAt: now.Unix(),
 	}
 
-	e, err := i.MemDBEntityByID(te.EntityID, true)
+	e, err := i.MemDBEntityByID(req.EntityID, true)
```

Comment:
@kalafut maybe add a nil check for e? not sure what the correct logic is if it is nil. Error? Don't attempt to template groups?

Reply:
Yes. I'm erroring out on e==nil now.